src/logic/contracts/safeContractErrors.ts

```diff
@@ -38,5 +38,6 @@ export const getContractErrorMessage = async ({
     return decodeMessage(contractOutput)
   } catch (err) {
     logError(Errors._817, err.message)
+    throw new CodedException(Errors._817, err.message)
```

Comment:
Just a thought because I'm not the biggest fan of multiple try-catches:
We could directly return error message with the error code and rename the function to `getContractError()`.
```suggestion
   return {code: Errors._817, message: err.message}
```

Reply:
IMHO throwing is more explicit and clean.

Reply:
I'm leaving the `throw new CodedException(...)` in this PR but we could discuss this point more broadly. I am including a bullet to discuss after the daily tomorrow.